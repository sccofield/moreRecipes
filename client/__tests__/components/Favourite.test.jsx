import React from 'react';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';
import { Favourite, mapDispatchToProps }
  from '../../src/components/Favourite';

describe('Favouritenvote component', () => {
  const props = {
    id: 2,
    favorites: [{
      userId: 2,
      recipeId: 3
    }
    ],
    userId: 2,
    favouriteRecipeActionCreator: jest.fn(() => Promise.resolve()),
  };

  it('should render the component', () => {
    const wrapper = shallow(<Favourite {...props} />);
    expect(wrapper.find('button').exists).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should show that the Downvote has been clicked', () => {
    const spy = sinon.spy(Favourite.prototype, 'favourite');
    const wrapper = shallow(<Favourite
      {...props}
    />);
    wrapper.find('.btn').simulate('click');
    // expect(spy.called).toBeTruthy();
  });
});

describe('conatiner functions', () => {
  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch))
      .toHaveProperty('favouriteRecipeActionCreator');

    const { favouriteRecipeActionCreator } = mapDispatchToProps(dispatch);
    favouriteRecipeActionCreator();
    expect(dispatch).toHaveBeenCalled();
  });
});
