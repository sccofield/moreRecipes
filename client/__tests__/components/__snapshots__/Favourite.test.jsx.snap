// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favouritenvote component should render the component 1`] = `
<a
  className="btn btn-outline-info m-1 favourite"
  id="favouriteButton"
>
  <i
    aria-hidden="true"
    className="fa fa-heart"
    onClick={[Function]}
  />
</a>
`;
